itest/litd_firewall_test.go

```diff
@@ -891,6 +943,416 @@ func testSessionLinking(net *NetworkHarness, t *harnessTest) {
+	// wrong (testing the policy rule), giving us back an error for both
+	// rule violations.
+	_, err = lndConn1.BatchOpenChannel(
+		ctx, &lnrpc.BatchOpenChannelRequest{
+			Channels: []*lnrpc.BatchOpenChannel{
+				{
+					NodePubkey:         charlie.PubKey[:],
+					LocalFundingAmount: 6_000_001,
+					MinHtlcMsat:        10,
+				},
+			},
+		}, caveatCreds,
+	)
+	require.ErrorContains(t, err, "invalid total capacity")
+	require.ErrorContains(t, err, "invalid min htlc msat amount")
+
+	// Now test a successful channel open, spending 6 Msat out of the 10
+	// Msat budget.
+	batchResp, err := lndConn1.BatchOpenChannel(
+		ctx, &lnrpc.BatchOpenChannelRequest{
+			Channels: []*lnrpc.BatchOpenChannel{
+				{
+					NodePubkey:         charlie.PubKey[:],
+					LocalFundingAmount: 6_000_000,
+					MinHtlcMsat:        1_000,
+				},
+			},
+		}, caveatCreds,
+	)
+	require.NoError(t, err)
+
+	assertChannelExistsAndClose := func(txIdHidden []byte,
+		outIdxHidden uint32) {
+
+		net.Miner.AssertNumTxsInMempool(1)
+		net.Miner.MineBlocks(6)
+
+		// The channel open response's txid is to be interpreted as a
+		// byte reveresed hash, which is used to check that the channel
+		// is present.
+		txHashHidden, err := chainhash.NewHash(txIdHidden)
+		require.NoError(t, err)
+
+		aliceFW := litrpc.NewFirewallClient(rawConn)
+
+		// Query Alice's privacy mapper to see what the real txid is.
+		chanPoint := fmt.Sprintf(
+			"%s:%d", txHashHidden.String(), outIdxHidden,
+		)
+		privMapResp, err := aliceFW.PrivacyMapConversion(
+			ctxm, &litrpc.PrivacyMapConversionRequest{
+				SessionId: sessResp.Session.Id,
+				Input:     chanPoint,
+			},
+		)
+		require.NoError(t, err)
+
+		// We assert that the channel exists using the real channel
+		// point.
+		txid, outputIndex, err := firewalldb.DecodeChannelPoint(
+			privMapResp.Output,
+		)
+		require.NoError(t, err)
+
+		tx, err := chainhash.NewHashFromStr(txid)
+		require.NoError(t, err)
+
+		outpoint := wire.OutPoint{
+			Hash:  *tx,
+			Index: outputIndex,
+		}
+		err = ht.lndHarness.AssertChannelExists(
+			ht.lndHarness.Alice, &outpoint,
+		)
+		require.NoError(t, err)
+
+		// Before doing any further tests, make sure to close out the
+		// channel should tests fail.
+		defer closeChannelAndAssert(ht, net, net.Alice,
+			&lnrpc.ChannelPoint{
+				FundingTxid: &lnrpc.ChannelPoint_FundingTxidStr{
+					FundingTxidStr: txid,
+				},
+				OutputIndex: outputIndex,
+			}, false)
+
+		// We check that a memo was added to the channel.
+		channels, err := net.Alice.ListChannels(
+			ctx, &lnrpc.ListChannelsRequest{},
+		)
+		require.NoError(t, err)
+
+		for _, channel := range channels.Channels {
+			if channel.ChannelPoint == outpoint.String() {
+				parts := strings.Split(channel.Memo, ":")
+				require.GreaterOrEqual(t, len(parts), 2)
+
+				// We expect the format of lndConnID-reqID.
+				parts = strings.Split(parts[0], "-")
+				require.Len(t, parts, 2)
+				require.Len(t, parts[0], terminal.LndConnIdLen)
```

Comment:
nit: I think we should add coverage to prove that the `reqID` (`parts[1]`) in the memo field varies for every made OpenChannelRequest.